pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -310,20 +310,7 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 						machine.Name, maxUnhealthyCount)
 					return reconcile.Result{Requeue: true}, nil
 				}
-				if !machine.GetDeletionTimestamp().IsZero() {
-					// Delete already initiated
-					return reconcile.Result{}, nil
-				}
-
-				if err := r.client.Delete(context.TODO(), machine); err != nil {
-					r.recorder.Eventf(machine, core.EventTypeWarning, "MachineDeletionFailed",
-						"Machine %v deletion failed: %v", machine.Name, err)
-					return reconcile.Result{}, err
-				}
-				log.Info("machine has been remediated by deletion", "name", machine.GetName())
-				r.recorder.Eventf(machine, core.EventTypeNormal, "MachineDeleted",
-					"Machine %v has been remediated by deleting the Machine object", machine.Name)
-				return reconcile.Result{}, nil
+				return reconcile.Result{}, r.deleteMachine(machine)
```

Comment:
Do we not need to validate if we have the expected private key when the machine is in `Running` phase ?

Reply:
Sorry, I see you are handling only `unconfigured` machines in this commit.